server/middlewares/incident.js

```diff
@@ -1,337 +1,316 @@
-        if (foundIncidentCreateBy) {
-            return res.status(409).json({
-                status: 409,
-                message: 'Incident already exists, consider updating it instead'
-            });
-        }
+    if (createdBy.length < 1) {
+      return res.status(400).json({
+        status: 400,
+        message: 'createdBy number starts from 1 and above'
+      });
+    }
+    if (createdBy === '') {
+      return res.status(400).json({
+        status: 400,
+        message: 'createdBy cannot be empty'
+      });
+    }
+    const foundIncidentCreateBy = db.incident.find(incident => incident.createdBy === createdBy);
+    if (foundIncidentCreateBy) {
+      return res.status(409).json({
+        status: 409,
+        message: 'Incident already exists, consider updating it instead'
+      });
+    }
 
-        //Type
-        if (type === undefined) {
-            return res.status(400).json({
-                status: 400,
-                message: 'Incident type cannot be undefined'
-            });
-        }
-        if (typeof type !== "string") {
-            return res.status(400).json({
-                status: 400,
-                message: 'Incident type should be a string'
-            });
-        }
-        if (type === '') {
-            return res.status(400).json({
-                status: 400,
-                message: 'Incident should have a type'
-            });
-        }
-         
-       
-        if (type !== 'red-flag') {
-            return res.status(400).json({
-                status: 400,
-                message: 'incident should have a type either red-flag or intervention'
-            })
-        }
-        // if (type !== 'intervention') {
-        //     return res.status(400).json({
-        //         status: 400,
-        //         message: 'incident should have a type either red-flag or intervention'
-        //     })
-        // }
-        //location
-        if (location === undefined) {
-            return res.status(400).json({
-                status: 400,
-                message: 'Incident location cannot be undefined'
-            });
-        }
-        if (typeof location !== "string") {
-            return res.status(400).json({
-                status: 400,
-                message: 'Incident location should be a string'
-            });
-        }
-        if (location === '') {
-            return res.status(400).json({
-                status: 400,
-                message: 'Incident should have a location'
-            });
-        }
-        const locationVerify = /^\d{1}\w{1}\s\d{1}\w{1}$/;
-        if (!locationVerify.test(location)) {
-            return res.status(400).json({
-                status: 400,
-                message: 'Invalid location  Input'
-            });
-        }
+    if (type === undefined) {
+      return res.status(400).json({
+        status: 400,
+        message: 'Incident type cannot be undefined'
```

Comment:
Apply the feedback about `undefined` here too.

Reply:
Implemented and resolved